File: backend/app/schemas/report.py
"""Report payload Pydantic models.

对齐 docs/specs/report-schema.md。任何字段变更必须同 PR 改 spec 文档。
"""

from typing import Literal

from pydantic import BaseModel, Field

Severity = Literal["high", "medium", "low"]
CategoryCode = Literal["H1", "H2", "H3", "H4", "H5", "H6", "H7", "H8", "H9", "H10"]


class Hazard(BaseModel):
    category_code: CategoryCode
    category_name: str
    description: str = Field(max_length=100)
    severity: Severity
    regulation: str = ""
    suggestion: str = Field(max_length=100)
    # 重大事故隐患（建质规〔2024〕5号）；命中即上报义务，前端红标区分。
    # 默认值保证向后兼容：旧 LLM 响应未带这两个字段时仍可解析。
    is_major: bool = False
    major_basis: str = ""


class ModelMeta(BaseModel):
    provider: Literal["claude_cli", "doubao", "fake"]
    model: str
    latency_ms: int = Field(ge=0)


class ReportPayload(BaseModel):
    inspection_id: str
    created_at: str
    plain_warning: str = Field(max_length=30)
    summary: str = Field(max_length=100)
    overall_severity: Severity
    hazards: list[Hazard]
    model_meta: ModelMeta

File: miniprogram/src/types/report.ts
/**
 * 报告 JSON Schema 的 TypeScript 镜像。
 *
 * 单一来源：`docs/specs/report-schema.md`
 * 后端对照：`backend/app/schemas/report.py`
 *
 * 任何字段增删 / 类型变更必须同 PR 改 spec + 后端 Pydantic 模型 + 本文件。
 * `tests/types/report.test.ts` 会读取 spec 的第一个 ```json 块做结构断言，
 * schema 漂移时该测试会挂。
 */

export type Severity = 'high' | 'medium' | 'low';

export type CategoryCode =
  | 'H1'
  | 'H2'
  | 'H3'
  | 'H4'
  | 'H5'
  | 'H6'
  | 'H7'
  | 'H8'
  | 'H9'
  | 'H10';

export interface Hazard {
  category_code: CategoryCode;
  category_name: string;
  description: string;
  severity: Severity;
  /** 可为空字符串；不允许编造规范条款。 */
  regulation: string;
  suggestion: string;
}

export interface ModelMeta {
  provider: 'claude_cli' | 'fake';
  model: string;
  latency_ms: number;
}

export interface ReportPayload {
  inspection_id: string;
  /** ISO 8601 UTC，例如 "2026-05-18T08:23:11Z"。 */
  created_at: string;
  /** 1-30 字口语化中文警示，顶部醒目卡片用。 */
  plain_warning: string;
  /** ≤ 100 字，专业总结，面向安全员。 */
  summary: string;
  /** 取 hazards[] 中最高 severity；空列表时为 "low"。 */
  overall_severity: Severity;
  hazards: Hazard[];
  model_meta: ModelMeta;
}
